executor/set_test.go

```diff
@@ -1607,3 +1607,25 @@ func TestInstanceScopeSwitching(t *testing.T) {
 	tk.MustExec("set tidb_enable_legacy_instance_scope = 0")
 	tk.MustGetErrCode("set tidb_general_log = 1", errno.ErrGlobalVariable)
 }
+
+func TestPreparePlanCacheValid(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+
+	tk.MustExec("SET GLOBAL tidb_enable_prepared_plan_cache = 1")
+	defer tk.MustExec("SET GLOBAL tidb_enable_prepared_plan_cache = 0")
+
+	tk.MustExec("SET GLOBAL tidb_prepared_plan_cache_size = 0")
+	tk.MustQuery("show warnings").Check(testkit.Rows(
+		"Warning 1292 Truncated incorrect tidb_prepared_plan_cache_size value: '0'"))
+	tk.MustExec("SET GLOBAL tidb_prepared_plan_cache_size = 2")
```

Comment:
Better to read the global variable and check its value again to ensure it has been set successfully.